presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -948,7 +950,8 @@ public boolean isImpersonationEnabled()
         if (refreshMillis.isPresent() && (expiresAfterWriteMillis.isEmpty() || expiresAfterWriteMillis.getAsLong() > refreshMillis.getAsLong())) {
             cacheBuilder = cacheBuilder.refreshAfterWrite(refreshMillis.getAsLong(), MILLISECONDS);
         }
-        cacheBuilder = cacheBuilder.maximumSize(maximumSize);
+        cacheBuilder = cacheBuilder.maximumSize(maximumSize)
+            .recordStats();
```

Comment:
please only records stats for caching metastore between transactions. Do not enable it for transactional caching metastore (the one which is created per each transaction).

Reply:
Done. I'm not using recordStats() when cache created via `memoizeMetastore` that used in transactions